utilities/navx.py

```diff
@@ -1,43 +1,41 @@
 import math
 
-import wpilib
 from robotpy_ext.common_drivers.navx.ahrs import AHRS
-from robotpy_ext.common_drivers.navx.registerio_spi import RegisterIO_SPI
+from wpilib.interfaces import PIDSource
 
-from utilities.functions import constrain_angle
 
+class NavX:
+    """Wrapper of robotpy navx to be inline with BNO055 api.
+    """
 
-class NavX(AHRS):
+    PIDSourceType = PIDSource.PIDSourceType
+
+    def __init__(self):
+        self.ahrs = AHRS.create_spi(update_rate_hz=100)
+        self.pidsource = self.PIDSourceType.kDisplacement
 
     def getAngle(self):
-        raw = super().getAngle()
-        return constrain_angle(-math.radians(raw))
+        """Get navx angle
+        returns:
+            angle in radians between -pi and +pi
+        """
+        raw = self.ahrs.getYaw()
+        return -math.radians(raw)
 
     def resetHeading(self):
-        self.reset()
+        self.ahrs.reset()
 
     def getHeadingRate(self):
-        return math.radians(-self.getRate())
-
-    # fix broken create_spi
-    @classmethod
-    def create_spi(cls, port=wpilib.SPI.Port.kMXP, spi_bitrate=None, update_rate_hz=None):
-        """Constructs the AHRS class using SPI communication, overriding the
-        default update rate with a custom rate which may be from 4 to 100,
-        representing the number of updates per second sent by the sensor.
+        return math.radians(-self.ahrs.getRate()) * 100  # short-term hack
```

Comment:
This should mention _why_ this hack exists, not just that it's a hack.

I'm not sure how long you think "short-term" is either :)